src/app/locator/qgsinbuiltlocatorfilters.cpp

```diff
@@ -298,6 +302,106 @@ void QgsActiveLayerFeaturesLocatorFilter::triggerResult( const QgsLocatorResult
   QgisApp::instance()->mapCanvas()->zoomToFeatureIds( layer, QgsFeatureIds() << id );
 }
 
+//
+// QgsAllLayersFeaturesLocatorFilter
+//
+
+QgsAllLayersFeaturesLocatorFilter::QgsAllLayersFeaturesLocatorFilter( QObject *parent )
+  : QgsLocatorFilter( parent )
+{
+  setUseWithoutPrefix( false );
+}
+
+QgsAllLayersFeaturesLocatorFilter *QgsAllLayersFeaturesLocatorFilter::clone() const
+{
+  return new QgsAllLayersFeaturesLocatorFilter();
+}
+
+void QgsAllLayersFeaturesLocatorFilter::prepare( const QString &string, const QgsLocatorContext & )
+{
+  if ( string.length() < 3 )
+    return;
+
+  const QMap<QString, QgsMapLayer *> layers = QgsProject::instance()->mapLayers();
+  for ( auto it = layers.constBegin(); it != layers.constEnd(); ++it )
+  {
+    QgsVectorLayer *layer = qobject_cast< QgsVectorLayer *>( it.value() );
+    if ( !layer || !layer->searchable() )
+      continue;
+
+    QgsExpression expression( layer->displayExpression() );
+    QgsExpressionContext context;
+    context.appendScopes( QgsExpressionContextUtils::globalProjectLayerScopes( layer ) );
+    expression.prepare( &context );
+
+    QgsFeatureRequest req;
+    req.setFlags( QgsFeatureRequest::NoGeometry );
```

Comment:
You could also set the attributes fetched to match those required by the expression, also better only set NoGeometry flag if the expression doesn't require it. (Someone maybe has a display expression which uses geometry?)

Reply:
done